streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -56,40 +56,42 @@ trait ZSink[-R, +E, +A0, -A, +B] { self =>
   /**
    * Operator alias for `zipRight`
    */
-  final def *>[R1 <: R, E1 >: E, A00 >: A0, A1 <: A, C](
-    that: ZSink[R1, E1, A00, A1, C]
-  )(implicit ev: A00 =:= A1, ev2: A1 =:= A00): ZSink[R1, E1, A00, A1, C] =
+  final def *>[R1 <: R, E1 >: E, C](
+    that: ZSink[R1, E1, A, C]
+  ): ZSink[R1, E1, A, C] =
     zip(that).map(_._2)
 
   /**
    * Operator alias for `zipLeft`
    */
-  final def <*[R1 <: R, E1 >: E, A00 >: A0, A1 <: A, C](
-    that: ZSink[R1, E1, A00, A1, C]
-  )(implicit ev: A00 =:= A1, ev2: A1 =:= A00): ZSink[R1, E1, A00, A1, B] =
+  final def <*[R1 <: R, E1 >: E, C](
+    that: ZSink[R1, E1, A, C]
+  ): ZSink[R1, E1, A, B] =
     zip(that).map(_._1)
 
   /**
    * Operator alias for `zip`
    */
-  final def <*>[R1 <: R, E1 >: E, A00 >: A0, A1 <: A, C](
-    that: ZSink[R1, E1, A00, A1, C]
-  )(implicit ev: A00 =:= A1, ev2: A1 =:= A00): ZSink[R1, E1, A00, A1, (B, C)] =
+  final def <*>[R1 <: R, E1 >: E, C](
+    that: ZSink[R1, E1, A, C]
+  ): ZSink[R1, E1, A, (B, C)] =
     self zip that
 
   /**
    * Operator alias for `orElse` for two sinks consuming and producing values of the same type.
    */
-  final def <|[R1 <: R, E1, B1 >: B, A00 >: A0, A1 <: A](
-    that: ZSink[R1, E1, A00, A1, B1]
-  )(implicit ev: A1 =:= A00): ZSink[R1, E1, A00, A1, B1] =
+  final def <|[R1 <: R, E1, B1 >: B](
+    that: ZSink[R1, E1, A, B1]
+  ): ZSink[R1, E1, A, B1] =
     (self orElse that).map(_.merge)
 
-  private[ZSink] final def ?[A00 >: A0, A1 <: A](implicit ev: A1 =:= A00): ZSink[R, Nothing, A00, A1, Option[B]] =
-    new ZSink[R, Nothing, A00, A1, Option[B]] {
+  final def optional = ?
+
+  final def ? : ZSink[R, Nothing, A, Option[B]] =
```

Comment:
It seems that we have lost the scaladoc when deleting the syntax classes.

Reply:
👍🏻